integration/qbft/scenarios/regular.go

```diff
@@ -57,115 +61,160 @@ func Regular(role spectypes.BeaconRole) *IntegrationTest {
+		for i := 1; i <= 4; i++ {
+			operatorIDIterator := spectypes.OperatorID(i)
+
+			expectedMsg = spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[operatorIDIterator], operatorIDIterator, &specqbft.Message{
+				MsgType:    specqbft.PrepareMsgType,
 				Height:     specqbft.FirstHeight,
 				Round:      specqbft.FirstRound,
 				Identifier: identifier[:],
-				Data:       proposalData,
-			}),
-			Decided:      true,
-			DecidedValue: consensusData,
-			ProposeContainer: &specqbft.MsgContainer{Msgs: map[specqbft.Round][]*specqbft.SignedMessage{
-				specqbft.FirstRound: {
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[1], 1, &specqbft.Message{
-						MsgType:    specqbft.ProposalMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       proposalData,
-					}),
-				},
-			}},
-			PrepareContainer: &specqbft.MsgContainer{Msgs: map[specqbft.Round][]*specqbft.SignedMessage{
-				specqbft.FirstRound: {
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[1], 1, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[2], 2, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[3], 3, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[4], 4, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-				},
-			}},
-			// NOTE: need to keep round keys and commit message data, there's a special check for CommitContainer
-			CommitContainer: &specqbft.MsgContainer{Msgs: map[specqbft.Round][]*specqbft.SignedMessage{
-				specqbft.FirstRound: {
-					&specqbft.SignedMessage{
-						Message: &specqbft.Message{
-							Data: commitData,
-						},
-					},
-				},
-			}},
-			RoundChangeContainer: &specqbft.MsgContainer{Msgs: map[specqbft.Round][]*specqbft.SignedMessage{}},
-		},
-		DecidedMessage: &specqbft.SignedMessage{
-			Message: &specqbft.Message{
+				Data:       prepareData,
+			})
+			if !isMessageExistInRound(expectedMsg, actual.State.PrepareContainer.Msgs[specqbft.FirstRound]) {
```

Comment:
@egorka-sereda 

If there're duplicated messages in the container (e.g. two from 1, one from 2, one from 3, one from 4), the test would pass, but needs to fail. 

Instead, could you please build a slice of 4 expected messages, sort them by the first signer, sort the actual result and then check if they are the same?

BTW, we might consider passing testing.T and using https://github.com/stretchr/testify because it has nice functions for assertions, which will remove a lot of code. For this case, https://pkg.go.dev/github.com/stretchr/testify/assert#Subset would be nice to use.

Reply:
@amir-blox Your thoughts about using `testify`?